@groeneai there is a failure of tidy build:
```
DEBUG:clang-tidy-cache.py:Calling real clang-tidy.
/ClickHouse/src/AggregateFunctions/Combinators/AggregateFunctionArray.h:44:13: error: multiple declarations in a single statement reduces readability [readability-isolate-declaration,-warnings-as-errors]
   44 |             String outer_params_str, nested_params_str;
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Fix it.

--- a/src/AggregateFunctions/Combinators/AggregateFunctionArray.h
+++ b/src/AggregateFunctions/Combinators/AggregateFunctionArray.h
@@ -41,7 +41,8 @@ public:
             /// parameter set. If this fires, it means some code path in
             /// AggregateFunctionFactory or a combinator wrapper lost/modified parameters.
             /// The diagnostic info below will identify the exact mismatch.
-            String outer_params_str, nested_params_str;
+            String outer_params_str;
+            String nested_params_str;
             for (const auto & p : parameters)
             {
                 if (!outer_params_str.empty()) outer_params_str += ", ";

--- a/src/AggregateFunctions/Combinators/AggregateFunctionArray.h
+++ b/src/AggregateFunctions/Combinators/AggregateFunctionArray.h
@@ -13,6 +13,7 @@ struct Settings;
 
 namespace ErrorCodes
 {
+    extern const int LOGICAL_ERROR;
     extern const int SIZES_OF_ARRAYS_DONT_MATCH;
     extern const int ILLEGAL_TYPE_OF_ARGUMENT;
 }
@@ -33,7 +34,32 @@ public:
         : IAggregateFunctionHelper<AggregateFunctionArray>(arguments, params_, createResultType(nested_))
         , nested_func(nested_), num_arguments(arguments.size())
     {
-        chassert(parameters == nested_func->getParameters());
+        if (parameters != nested_func->getParameters())
+        {
+            /// This invariant should always hold: the Array combinator does not transform
+            /// parameters, so the wrapped function must have been created with the same
+            /// parameter set. If this fires, it means some code path in
+            /// AggregateFunctionFactory or a combinator wrapper lost/modified parameters.
+            /// The diagnostic info below will identify the exact mismatch.
+            String outer_params_str, nested_params_str;
+            for (const auto & p : parameters)
+            {
+                if (!outer_params_str.empty()) outer_params_str += ", ";
+                outer_params_str += p.dump();
+            }
+            for (const auto & p : nested_func->getParameters())
+            {
+                if (!nested_params_str.empty()) nested_params_str += ", ";
+                nested_params_str += p.dump();
+            }
+            throw Exception(ErrorCodes::LOGICAL_ERROR,
+                "AggregateFunctionArray: parameters mismatch between Array wrapper '{}' "
+                "and nested function '{}'. Wrapper has {} parameter(s): [{}], "
+                "nested function has {} parameter(s): [{}]",
+                getName(), nested_func->getName(),
+                parameters.size(), outer_params_str,
+                nested_func->getParameters().size(), nested_params_str);
+        }
         for (const auto & type : arguments)
             if (!isArray(type))
                 throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "All arguments for aggregate function {} must be arrays", getName());

